src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/ProjectTestFixture.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class ProjectTestFixture : IAsyncLifetime
```

Comment:
This fixture is similar to what I moved from `ProjectDirectory` and `StaticWebAssetsIntegrationTest` (`PackageTestProjectsFixture` in particular) but
1. moves global.json and the TestPackageRestoreSource folder up into `$env:Temp\Razor` (one level up in most cases)
2. adds a NuGet.config file to `$env:Temp\Razor`
3. only does setup once per test run
4. removes the entire `$env:Temp\Razor` folder when done (by default)
5. uses an `IMessageSink` instead of an `ITestOutputHelper` because the first is all that's available in a test fixture